packages/kotti-ui/source/kotti-comment/KtComment.vue

```diff
@@ -95,75 +104,74 @@ export default {
 		KtCommentInput,
 	},
 	props: makeProps(KottiComment.propsSchema),
-	data() {
-		return {
-			showInlineReply: false,
-			replyToText: null,
-			replyToUserId: Number,
-			isInlineEdit: false,
-			inlineMessageValue: '',
+	setup(props, { emit }) {
+		const isInlineEdit = ref(false)
+		const showInlineReply = ref(false)
+		const inlineMessageValue = ref<string | null>(null)
+		const userBeingRepliedTo = ref<UserData | null>(null)
+
+		const handleDelete = (commentId: number | string, isInline?: boolean) => {
+			emit('delete', {
+				id: commentId,
+				parentId: isInline ? props.id : null,
+			})
 		}
-	},
-	computed: {
-		styleObject() {
-			return {
-				avatar: true,
-				'avatar-selected': this.selected,
-			}
-		},
-		inlineMessage() {
-			return this.inlineMessageValue || this.message
-		},
-		actionOptions() {
-			const options = []
-			if (this.isEditable)
-				options.push({
-					label: 'Edit',
-					onClick: () => {
-						this.inlineMessageValue = this.inlineMessage
-						this.isInlineEdit = true
-					},
-					type: 'EDIT',
-				})
-			if (this.isDeletable)
-				options.push({
-					label: 'Delete',
-					onClick: () => this.handleDelete(this.id),
-					type: 'DELETE',
+		return {
+			actionOptions: computed<Array<KottiButton.Props>>(() =>
+				compact([
+					props.isEditable
+						? {
+								label: 'Edit',
+								onClick: () => {
+									if (inlineMessageValue.value === null)
+										inlineMessageValue.value = props.message
+									isInlineEdit.value = true
+								},
+								type: KottiButton.Type.PRIMARY,
+						  }
+						: null,
+					props.isDeletable
+						? {
+								label: 'Delete',
+								onClick: () => handleDelete(props.id),
+								type: KottiButton.Type.DANGER,
+						  }
+						: null,
+				]),
+			),
+			cancelInlineEdit: () => {
+				inlineMessageValue.value = null
+				isInlineEdit.value = false
+			},
+			handleDelete,
```

Comment:
can you explain why we have BOTH `delete` on the options of the comment and `inline`

the options on the comment are controlled by the user-app

e.g. in AMPI, depending on user permissions the `delete` option is disabled if the user doesn't have deletion rights


So my question is: is this `inlineDelete` controlled??
looks to me like a duplicate feature that most likely is not noticed by the user-app

Reply:
I  am not 100% I get your question. If a comment reply should be deleted, it emits a delete, since The KtComment is always its parent, it re-emits, so it can be handled by the user app. AKA, all `delete/edit/submit` events are emitted by `KtComment` and are processed by whoever implements this.